Fix test test_numpy_torch_operators as the operator matrixmultiply
issues a ValueError when called on two scalars.

Author: Jakob Schiotz, Tech. Univ. Denmark
Email: schiotz@fysik.dtu.dk
Date: 8. November 2023

Stacktrace:

==================================================================== FAILURES =====================================================================
______________________________________________________ MiscTests.test_numpy_torch_operators _______________________________________________________
Traceback (most recent call last):
  File "/home/modules/software/Python/3.11.3-GCCcore-12.3.0/lib/python3.11/unittest/case.py", line 57, in testPartExecutor
    yield
  File "/home/modules/software/Python/3.11.3-GCCcore-12.3.0/lib/python3.11/unittest/case.py", line 623, in run
    self._callTestMethod(testMethod)
  File "/home/modules/software/Python/3.11.3-GCCcore-12.3.0/lib/python3.11/unittest/case.py", line 579, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/tmp/eb-19tah_22/tmplupwseoe/lib/python3.11/site-packages/torch/testing/_internal/common_utils.py", line 2388, in wrapper
    method(*args, **kwargs)
  File "/scratch/schiotz/eb_build/PyTorch/2.1.0/foss-2023a/pytorch-v2.1.0/test/dynamo/test_misc.py", line 1326, in test_numpy_torch_operators
    result = op(t1, t2)
             ^^^^^^^^^^
ValueError: inplace matrix multiplication requires the first operand to have at least one and the second at least two dimensions.

To execute this test, run the following from the base repo dir:
     python test/dynamo/test_misc.py -k test_numpy_torch_operators

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
frames [('total', 9), ('ok', 9)]
stats [('calls_captured', 9), ('unique_graphs', 9)]
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
frames [('total', 9), ('ok', 9)]
stats [('calls_captured', 9), ('unique_graphs', 9)]
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------
frames [('total', 9), ('ok', 9)]
stats [('calls_captured', 9), ('unique_graphs', 9)]
============================================================= short test summary info =============================================================
FAILED [0.0735s] dynamo/test_misc.py::MiscTests::test_numpy_torch_operators - ValueError: inplace matrix multiplication requires the first opera...
========================================== 1 failed, 294 passed, 9 skipped, 2 xfailed, 2 rerun in 22.40s ==========================================

diff --git a/test/dynamo/test_misc.py b/test/dynamo/test_misc.py
index c5ed05a..5c6fbcb 100644
--- a/test/dynamo/test_misc.py
+++ b/test/dynamo/test_misc.py
@@ -1324,7 +1324,7 @@ class MiscTests(torch._dynamo.test_case.TestCase):
             try:
                 # TODO try a bit harder
                 result = op(t1, t2)
-            except (RuntimeError, TypeError, IndexError):
+            except (RuntimeError, TypeError, IndexError, ValueError):
                 continue
             cnts = torch._dynamo.testing.CompileCounter()
             opt_fn = torch._dynamo.optimize(cnts)(fn)
